pkg/skaffold/deploy/labeller.go

```diff
@@ -19,40 +19,50 @@ package deploy
 import (
 	"fmt"
 
+	"github.com/google/uuid"
+
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/version"
 )
 
 const (
-	K8ManagedByLabelKey = "app.kubernetes.io/managed-by"
-	UnknownVersion      = "unknown"
-	Empty               = ""
+	K8sManagedByLabelKey = "app.kubernetes.io/managed-by"
+	UuidRunIdLabel       = "skaffold.dev/run-id"
+	unknownVersion       = "unknown"
+	empty                = ""
 )
 
-// DefaultLabeller adds K9 style managed-by label
+// DefaultLabeller adds K8s style managed-by label and a run-specific UUID label
 type DefaultLabeller struct {
 	version string
+	uuid    string
 }
 
 func NewLabeller(verStr string) *DefaultLabeller {
-	if verStr == Empty {
+	if verStr == empty {
 		verStr = version.Get().Version
 	}
-	if verStr == Empty {
-		verStr = UnknownVersion
+	if verStr == empty {
+		verStr = unknownVersion
 	}
 	return &DefaultLabeller{
 		version: verStr,
+		uuid:    uuid.New().String(),
 	}
```

Comment:
can we make sure somehow that this is _guaranteed_ to run only once? i.e. with a `Once`. With the current design if we accidentally create multiple labellers this could result in multiple run ids.

Reply:
Good idea! The perfect use-case for a `Once` :)